src/messaging/ExchangeContext.cpp

```diff
@@ -157,8 +157,11 @@ CHIP_ERROR ExchangeContext::SendMessageImpl(uint16_t protocolId, uint8_t msgType
         mReliableMessageContext.SetAckPending(false);
 
 #if !defined(NDEBUG)
-        ChipLogProgress(ExchangeManager, "Piggybacking Ack for MsgId:%08" PRIX32 " with msg",
-                        mReliableMessageContext.mPendingPeerAckId);
+        if (!sendFlags.Has(SendMessageFlags::kNoAutoRequestAck))
```

Comment:
This doesn't make sense to me.  Why does it matter whether we're _requesting_ an ack, for purposes of logging this?  We're definitly piggybacking an ack here.

Reply:
No, we are not piggybacking the ack if the responder ack with a standalong ack message.  Without this check, "Piggybacking Ack" will always be printed and you will see the following log during standalong ack test. 

CHIP:EM: Sending solitary ack for MsgId:00000000
CHIP:EM: Piggybacking Ack for MsgId:00000000 with msg
CHIP:IN: Secure message was encrypted: Msg ID 0

Current test case use keyword "Piggybacking " and "solitary" to verify if ack is sent by Piggybacking or over a standalong ack message, so the test will fail here.

Reply:
> No, we are not piggybacking the ack if the responder ack with a standalong ack message

Oh, is this test trying to detect whether the message being sent is a standalone ack, which also go through this codepath?

Testing that would make sense, and in that case we should not log the "piggybacking" thing, but `kNoAutoRequestAck` doesn't mean that as far as I can tell.  For example, that flag is set for `MsgCounterSyncReq` messages too, no? If we want to test for standalone ack, shouldn't we test the payload header for the corresponding message type?

Reply:
MsgCounterSyncReq/MsgCounterSyncResp do not use CRMP, so it won't fall under this case.  

 if (mReliableMessageContext.HasPeerRequestedAck())
{
 .......
}

For the CRMP messages, only Standalong ack set flag kNoAutoRequestAck.

Reply:
Acked, I think use a dedicated Standalong ack flag to suppress "Piggybacking Ack" could cover all future corner cases.